server/mux.go

```diff
@@ -67,127 +67,137 @@ func NewMux(opts MuxOpts, service Service) http.Handler {
+	router.GET("/chronograf/v1/dashboards/:id/cells", EnsureViewer(service.DashboardCells))
+	router.POST("/chronograf/v1/dashboards/:id/cells", EnsureEditor(service.NewDashboardCell))
 
-	router.GET("/chronograf/v1/dashboards/:id/cells/:cid", service.DashboardCellID)
-	router.DELETE("/chronograf/v1/dashboards/:id/cells/:cid", service.RemoveDashboardCell)
-	router.PUT("/chronograf/v1/dashboards/:id/cells/:cid", service.ReplaceDashboardCell)
+	router.GET("/chronograf/v1/dashboards/:id/cells/:cid", EnsureViewer(service.DashboardCellID))
+	router.DELETE("/chronograf/v1/dashboards/:id/cells/:cid", EnsureEditor(service.RemoveDashboardCell))
+	router.PUT("/chronograf/v1/dashboards/:id/cells/:cid", EnsureEditor(service.ReplaceDashboardCell))
 	// Dashboard Templates
-	router.GET("/chronograf/v1/dashboards/:id/templates", service.Templates)
-	router.POST("/chronograf/v1/dashboards/:id/templates", service.NewTemplate)
+	router.GET("/chronograf/v1/dashboards/:id/templates", EnsureViewer(service.Templates))
+	router.POST("/chronograf/v1/dashboards/:id/templates", EnsureEditor(service.NewTemplate))
 
-	router.GET("/chronograf/v1/dashboards/:id/templates/:tid", service.TemplateID)
-	router.DELETE("/chronograf/v1/dashboards/:id/templates/:tid", service.RemoveTemplate)
-	router.PUT("/chronograf/v1/dashboards/:id/templates/:tid", service.ReplaceTemplate)
+	router.GET("/chronograf/v1/dashboards/:id/templates/:tid", EnsureViewer(service.TemplateID))
+	router.DELETE("/chronograf/v1/dashboards/:id/templates/:tid", EnsureEditor(service.RemoveTemplate))
+	router.PUT("/chronograf/v1/dashboards/:id/templates/:tid", EnsureEditor(service.ReplaceTemplate))
 
 	// Databases
-	router.GET("/chronograf/v1/sources/:id/dbs", service.GetDatabases)
-	router.POST("/chronograf/v1/sources/:id/dbs", service.NewDatabase)
+	router.GET("/chronograf/v1/sources/:id/dbs", EnsureViewer(service.GetDatabases))
+	router.POST("/chronograf/v1/sources/:id/dbs", EnsureEditor(service.NewDatabase))
 
-	router.DELETE("/chronograf/v1/sources/:id/dbs/:dbid", service.DropDatabase)
+	router.DELETE("/chronograf/v1/sources/:id/dbs/:dbid", EnsureAdmin(service.DropDatabase))
 
 	// Retention Policies
-	router.GET("/chronograf/v1/sources/:id/dbs/:dbid/rps", service.RetentionPolicies)
-	router.POST("/chronograf/v1/sources/:id/dbs/:dbid/rps", service.NewRetentionPolicy)
+	router.GET("/chronograf/v1/sources/:id/dbs/:dbid/rps", EnsureViewer(service.RetentionPolicies))
+	router.POST("/chronograf/v1/sources/:id/dbs/:dbid/rps", EnsureEditor(service.NewRetentionPolicy))
 
-	router.PUT("/chronograf/v1/sources/:id/dbs/:dbid/rps/:rpid", service.UpdateRetentionPolicy)
-	router.DELETE("/chronograf/v1/sources/:id/dbs/:dbid/rps/:rpid", service.DropRetentionPolicy)
+	router.PUT("/chronograf/v1/sources/:id/dbs/:dbid/rps/:rpid", EnsureEditor(service.UpdateRetentionPolicy))
+	router.DELETE("/chronograf/v1/sources/:id/dbs/:dbid/rps/:rpid", EnsureEditor(service.DropRetentionPolicy))
```

Comment:
Should this be EnsureAdmin if dropdatabase is?

Reply:
Are these defaults for Telegraf, for instance, it'll be Editor.